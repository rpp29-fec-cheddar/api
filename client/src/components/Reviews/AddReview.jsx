import React, {useState} from 'react';
import ReviewForm from './ReviewForm.jsx';

const AddReview = (props) => {
  const [show, setShow] = useState(false);
  return (
    <div className="addReview">
      <button className="addReview" onClick={ () => setShow(true) }>Add A Review</button>
      <ReviewForm
        name={props.name}
        productId={props.productId}
        characteristics={props.characteristics}
        show={show}
        onClose={ () => setShow(false) } />
    </div>
  )
}

export default AddReview;