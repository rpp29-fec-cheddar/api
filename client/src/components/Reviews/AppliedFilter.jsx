import React from 'react';

const AppliedFilter = (props) => {
  let filterNumber = props.filterNumber;

  return (
    <span
      className="appliedFilter"
      onClick={(e) => props.recordClick(e)}>
      {filterNumber} Stars<br></br>
    </span>
  )
}

export default AppliedFilter;