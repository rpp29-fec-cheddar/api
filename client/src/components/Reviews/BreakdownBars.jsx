import React from 'react';


class BreakdownBars extends React.Component {
  constructor() {
    super();
    this.state = {

    }
    //binding
  }

  //handleClick
    //will display reviews of that many stars when clicked
      //additive
      //toggles


  render() {
    // map the breakdowns for each line so that the amount of stars is listed (make this clickable),
    // then the graphics bar,
    // then the total number of reviews with that particular rating
    return (
      <div>
        null
        {/* ex. 5 Stars ____________----- (2) */}
        {/* must have onClick handler for 5star, 4star, 3star, 2star, 1star */}
      </div>
    )
  }
}



export default BreakdownBars;