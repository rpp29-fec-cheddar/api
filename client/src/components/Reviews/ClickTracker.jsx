import React from 'react';
import $ from 'jquery';

class ClickTracker extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      interactions: []
    }
    this.recordClick = this.recordClick.bind(this)
  }

  recordClick(event) {
    let newInteractions = this.state.interactions.slice();
    if (newInteractions.length > 5) {
      console.log('moving into AJAX call')
      $.ajax({
        type: 'POST',
        url: '/reviews/interactions',
        data: {interactions: newInteractions},
        success: () => {
          this.setState({
            interactions: []
          })
        },
        error: (err) => {
          console.log('ERROR in recordClick post!: ', err)
        }
      });
    }
    let element = event.target.className;
    let isRecorded = {
      element: element,
      widget: 'Reviews Widget',
      time: new Date(Date.now(event.target.timeStamp))
    }
    newInteractions.push(isRecorded);
    this.setState({
      interactions: newInteractions
    });
  }

  render() {
    let renderProps = {
      recordClick: this.recordClick
    }
    if (typeof this.props.children === 'function') {
      return this.props.children(renderProps);
    } else {
      return this.props.children
    }
  }
}

export default ClickTracker;