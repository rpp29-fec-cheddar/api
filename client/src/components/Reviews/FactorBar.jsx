import React from 'react';

const FactorBar = (props) => {
  let factorValue = (props.value / 5) * 13;
  let leftDescriptor,
    centerDescriptor,
    rightDescriptor;

  if (props.name === 'Size') {
    leftDescriptor = 'Too small';
    centerDescriptor = 'Perfect';
    rightDescriptor = 'Too large';
  }

  if (props.name === 'Width') {
    leftDescriptor = 'Too Narrow';
    centerDescriptor = 'Perfect';
    rightDescriptor = 'Too Wide'
  }

  if (props.name === 'Comfort') {
    leftDescriptor = 'Uncomfortable';
    centerDescriptor = 'Ok';
    rightDescriptor = 'Perfect'
  }

  if (props.name === 'Quality') {
    leftDescriptor = 'Poor';
    centerDescriptor = 'What I Expected';
    rightDescriptor = 'Perfect'
  }

  if (props.name === 'Length') {
    leftDescriptor = 'Runs Tight';
    centerDescriptor = 'Perfect';
    rightDescriptor = 'Runs Long';
  }

  if (props.name === 'Fit') {
    leftDescriptor = 'Tight';
    centerDescriptor = 'Perfect';
    rightDescriptor = 'Loose';
  }

  return (
    <div>
      <div className="factorName">{props.name}</div>
      <div className="factorContainer">
        <div className="factorBar">
          <div className="factorBreakLeft"></div>
          <div className="factorBreakRight"></div>
          <div className="factorPoint" style={{'marginLeft': `${factorValue}em`}}></div>
          <div className="descriptors">
            <div className="leftDescriptor">{leftDescriptor}</div>
            <div className="centerDescriptor">{centerDescriptor}</div>
            <div className="rightDescriptor">{rightDescriptor}</div>
          </div>
        </div>
        <br></br>
        <br></br>
      </div>
    </div>
  )
}

export default FactorBar;
