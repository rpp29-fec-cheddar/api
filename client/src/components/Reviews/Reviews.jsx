import React from 'react';
import $ from 'jquery';
import Stars from './Stars.jsx';
import Sort from './Sort.jsx';
import RatingBreakdown from './RatingBreakdown.jsx';
import FactorsBreakdown from './FactorsBreakdown.jsx';
import ReviewForm from './ReviewForm.jsx';

class Reviews extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      numOfRatings: 0,
      changed: false
    }
  }

  componentDidUpdate(prevProps) {
    if (this.props !== prevProps) {
      this.setState({
        numOfRatings: Object.keys(this.props.ratings).length,
        changed: true
      })
    }
  }

  // click handler for more Reviews button

  // click handler for add a review button

  render() {
    if (this.state.changed) {
      return (
        <div>
          <h1>Reviews</h1>
          <Stars
            renderStars={this.props.renderStars}
            averageRating={this.props.averageRating}
            numOfRatings={this.state.numOfRatings} />
          <br></br>
          <RatingBreakdown
            ratings={this.props.ratings}
            recommended={this.props.recommended} />
          <br></br>
          <br></br>
          <FactorsBreakdown characteristics={this.props.characteristics} />
          <br></br>
          <Sort
            helpfulReviews={this.props.helpfulReviews}
            newestReviews={this.props.newestReviews}
            recommended={this.props.recommended}
            renderStars={this.props.renderStars} />
          {/*
          <br></br>
          <ReviewTiles />
          more reviews button with onClick
          add a review button with onClick
          */}
        </div>
      )
    } else {
      return null;
    }
  }
}

export default Reviews;
