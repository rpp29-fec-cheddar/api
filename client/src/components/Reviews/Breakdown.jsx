import React from 'react';
import BreakdownBars from './BreakdownBars.jsx';


class Breakdown extends React.Component {
  constructor() {
    super();
    this.state = {
      reviews: [],
      percentage: 0
    }
    //binding
  }

  //ajax request to get array of reviews and percentage recommended
    //send to 'http://localhost:4000/reviews/reviews',


  render() {
    return(
      <div>
        {/* {send reviews data to be mapped to BreakdowBars} */}
        {/* product breakdown form will go here */}

        {/* The percentage of reviews that ‘recommend’
        the product will be displayed below the breakdown. */}
      </div>
    )
  }


}

export default Breakdown;