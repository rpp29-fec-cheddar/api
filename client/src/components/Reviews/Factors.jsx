import React from 'react';
import FactorBar from './FactorBar.jsx';
import $ from 'jquery';

class Factors extends React.Component {
  constructor() {
    super();
    this.state = {
      characteristics: []
      //characteristics will be an array of objects
    }
    //binding
    this.componentDidMount = this.componentDidMount.bind(this);
  }

  componentDidMount() {

  }

  render() {
    return (
      <div>
        {/* Each characteristic will be mapped out to FactorBar.jsx*/}
      </div>
    )
  }

}


export default Factors;