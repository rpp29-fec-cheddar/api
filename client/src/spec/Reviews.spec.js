/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import App from '../app.jsx';
import Reviews from '../components/Reviews/Reviews.jsx';

describe('Reviews Component', () => {
  test('Test that Reviews Component renders', () => {
    const component = renderer.create(<Reviews ></Reviews>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});





