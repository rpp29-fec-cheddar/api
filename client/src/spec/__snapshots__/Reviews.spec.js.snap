// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews Component Test that Reviews Component renders 1`] = `
<div>
  <div>
    <strong>
      !!!Average Rating and Stars Here!!!
    </strong>
    <h1>
      0
    </h1>
    <div>
      <div
        className="star-box"
      >
        <div
          className="star-filled-in"
          style={
            Object {
              "width": "0px",
            }
          }
        >
          <img
            alt="stars alt"
            className="star-outline"
            src="star.png"
          />
        </div>
      </div>
      <div
        className="star-box"
      >
        <div
          className="star-filled-in"
          style={
            Object {
              "width": "0px",
            }
          }
        >
          <img
            alt="stars alt"
            className="star-outline"
            src="star.png"
          />
        </div>
      </div>
      <div
        className="star-box"
      >
        <div
          className="star-filled-in"
          style={
            Object {
              "width": "0px",
            }
          }
        >
          <img
            alt="stars alt"
            className="star-outline"
            src="star.png"
          />
        </div>
      </div>
      <div
        className="star-box"
      >
        <div
          className="star-filled-in"
          style={
            Object {
              "width": "0px",
            }
          }
        >
          <img
            alt="stars alt"
            className="star-outline"
            src="star.png"
          />
        </div>
      </div>
      <div
        className="star-box"
      >
        <div
          className="star-filled-in"
          style={
            Object {
              "width": "0px",
            }
          }
        >
          <img
            alt="stars alt"
            className="star-outline"
            src="star.png"
          />
        </div>
      </div>
    </div>
    <h4>
      0
       reviews
    </h4>
  </div>
  <br />
  <div>
    <strong>
      !!!Rating Breakdown Here!!!
    </strong>
    <br />
    5 stars: 
    0
     
    <div>
      <strong>
        !!!Rating Bar Here!!!
      </strong>
    </div>
     
    <br />
    4 stars: 
    0
     
    <div>
      <strong>
        !!!Rating Bar Here!!!
      </strong>
    </div>
     
    <br />
    3 stars: 
    0
     
    <div>
      <strong>
        !!!Rating Bar Here!!!
      </strong>
    </div>
     
    <br />
    2 stars: 
    0
     
    <div>
      <strong>
        !!!Rating Bar Here!!!
      </strong>
    </div>
     
    <br />
    1 stars: 
    0
     
    <div>
      <strong>
        !!!Rating Bar Here!!!
      </strong>
    </div>
     
    <br />
    <h3>
      0
      % of reviews recommend this product
    </h3>
  </div>
  <br />
  <div>
    <strong>
      !!!Factors Breakdown Here!!!
    </strong>
    <div>
      <strong>
        !!!Factor Bar Here!!!
      </strong>
    </div>
    <div>
      <strong>
        !!!Factor Bar Here!!!
      </strong>
    </div>
  </div>
  <br />
  <div>
    <strong>
      !!!Sort Dropdown Here!!!
    </strong>
  </div>
  <br />
  <div>
    <strong>
      !!!Review Tiles Here!!!
    </strong>
  </div>
</div>
`;
